File: backend/src/CreditFinancialOrganization.Domain/Entities/Payments/Payment.cs
using CreditFinancialOrganization.Domain.Entities.Loans;
using CreditFinancialOrganization.Domain.Enums;

namespace CreditFinancialOrganization.Domain.Entities.Payments;

public class Payment : BaseEntity
{
    public Guid LoanId { get; set; }
    public DateTime Date { get; set; }
    public decimal Amount { get; set; }
    public PaymentStatus Status { get; set; }
    public PaymentMethod PaymentMethod { get; set; }

    public Loan Loan { get; set; } = null!;
}

File: frontend/src/app/shared/models/payments/payment.ts
import { PaymentMethod } from "../../enums/payment-method";
import { PaymentStatus } from "../../enums/payment-status";
import { Loan } from "../loans/loan";

export interface Payment {
    id: string;
    loan: Loan;
    date: Date;
    amount: number;
    status: PaymentStatus;
    paymentMethod: PaymentMethod;
}